pebblo/app/service/prompt_service.py

```diff
@@ -32,20 +32,22 @@ def _fetch_retrieval_context_data(self):
         """
         logger.debug("Retrieving retrieval context details from input data")
         # Fetching retrieval context details
-        retrieval_context = {}
+        retrieval_context_output = []
         context_data = self.data.get("context")
-        fields_validator(context_data, ["retrieved_from", "doc", "vector_db"])
         if context_data:
-            retrieval_context = RetrievalContext(
-                retrieved_from=context_data.get("retrieved_from"),
-                doc=context_data.get("doc"),
-                vector_db=context_data.get("vector_db"),
-            )
+            for data in context_data:
```

Comment:
Problem here is same as loader/doc API, where we have multiple snippets in the incoming request and we process it and send back response for all the snippets. It is possible that some context object gets skipped due to some exception. There is no way client would know if some context is missed.
In loader/doc we have implemented indexing.

Reply:
We can handle it in follow up PR.